python/test/unit/geometry/test_bounding_box_tree.py

```diff
@@ -496,3 +497,82 @@ def test_surface_bbtree_collision(dtype):
 
     collisions = compute_collisions_trees(bbtree1, bbtree2)
     assert len(collisions) == 1
+
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_determine_point_ownership(dtype):
+    """Find point owners (ranks and cells) using bounding box trees + global communication
+    and compare to point ownership data results."""
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+
+    tdim = 2
+    num_cells_side = 4
+    h = dtype(1.0 / num_cells_side)
+    mesh = create_unit_square(MPI.COMM_WORLD,
+                              num_cells_side, num_cells_side,
+                              CellType.quadrilateral, dtype=dtype)
+    cell_map = mesh.topology.index_map(tdim)
+
+    def test_po_against_bb_tree(mesh,points,cells):
+        cells = np.array([cell for cell in cells if cell < cell_map.size_local])
+        num_points = points.shape[0]
+        po = determine_point_ownership(mesh,points,cells)
+        tree = bb_tree(mesh, mesh.topology.dim, cells)
+        assert(mesh.topology.cell_type in [CellType.quadrilateral,CellType.hexahedron])
+        # Since box elements are considered, bounding box collisions correspond
+        # to the actual collisions
+        tree_col = compute_collisions_points(tree, points)
+        local_owners = np.zeros(2*num_points,dtype=np.int32) # rankwise data before Allreduce
+        owners = local_owners.copy()
+        for ipoint in range(num_points):
+            cell = tree_col.links(ipoint)
+            if len(cell) > 0:
+                local_owners[2*ipoint] = rank
+                local_owners[2*ipoint+1] = cell[0]
+        comm.Allreduce(local_owners, owners, op=MPI.SUM)
```

Comment:
```suggestion
        comm.Allreduce(local_owners, owners, op=MPI.SUM)
```
The current test has some unwritten assumptions.

1. All points in the input `points` are the same on all processes and in the same order
2. An input point can only be found on one process (i.e.) it is in the interior of a single cell and not on a facet/edge/vertex

If we want to keep these assumptions, then this should be specified when constructing `local_owners`, as it then make sense to use `np.zeros`, as the sum will not accumulate ranks or cell indices.
However,  shouldn't `local_owners[2*i+1]` use global cell index rather than a local index?

Reply:
Yes, 1 and 2 are indeed assumed. I will add a comment.
Since PointOwnershipData stores local cell indices, we use local indices rather than global indices for local_owners[2*i+1].